Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -371,125 +388,142 @@ final class ExecuteCommandTests: XCTestCase {
 
     let options = SourceKitLSPOptions.testDefault(experimentalFeatures: [.showMacroExpansions])
 
-    for peekDocuments in [false, true] {
-      let project = try await SwiftPMTestProject(
-        files: files,
-        manifest: SwiftPMTestProject.macroPackageManifest,
-        capabilities: ClientCapabilities(experimental: ["workspace/peekDocuments": .bool(peekDocuments)]),
-        options: options,
-        enableBackgroundIndexing: true
-      )
-
-      let (uri, positions) = try project.openDocument("MyMacroClient.swift")
-
-      let positionMarkersToBeTested = [
-        (start: "1️⃣", end: "1️⃣"),
-        (start: "2️⃣", end: "2️⃣"),
-        (start: "1️⃣", end: "3️⃣"),
-        (start: "2️⃣", end: "3️⃣"),
-      ]
-
-      for positionMarker in positionMarkersToBeTested {
-        let args = ExpandMacroCommand(
-          positionRange: positions[positionMarker.start]..<positions[positionMarker.end],
-          textDocument: TextDocumentIdentifier(uri)
+    for getReferenceDocument in [false, true] {
+      for peekDocuments in [false, true] {
```

Comment:
I’m wondering if it’s worth saving a nesting level by doing

```suggestion
    for (getReferenceDocument, peekDocuments) in [(false, false), (false, true), (true, false), (true, true)] {
```

What do you think? There’s definitely a trade-off here.

Reply:
I don't like the nesting too. I wish there was some function that goes like: `Bool.truthTableBase(variablesCount: 2)` which generates the true/false pattern for the given number of variables, It will also be less error-prone because we won't be hard coding the values, and will be more readable. That will then allow us to write the following:
```
    for (getReferenceDocument, peekDocuments) in Bool.truthTableBase(variablesCount: 2) {  
```
Even if any new test case needs similar looping with more boolean variables, It can go like:
```
    for (bool1, bool2, bool3, ..., boolN) in Bool.truthTableBase(variablesCount: N) {
```

But maybe this is over-engineering and just hard-coding the values would be fine for now.

Reply:
Or this could be a neat candidate for `SwiftExtensions` package 🤔

Reply:
I think the natural think to implement in `SwiftExtensions` would be a cartesian product, so that you would have

```swift
for (getReferenceDocument, peekDocuments) in cartesionProduct([true, false], [true, false]) {
```

Reply:
I wanted to write a very general `cartesianProduct` function, but seems like we will lose the ability to unpack the values since it returns an array.

Perhaps, parameter packs can be used in place of variadic parameters to solve this. I tried it out of our project with Swift 6, but am slightly confused with the implementation.

And even then, I am using Swift 5.10 for our regular workflow.

For now, I guess this would suffice. When the project moves to Swift 6 completely, then we can update the function to use parameter packs maybe.